aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsFluentClientDecorator.kt

```diff
@@ -113,15 +114,20 @@ private class AwsFluentClientExtensions(private val types: Types) {
                     let retry_config = conf.retry_config.as_ref().cloned().unwrap_or_default();
                     let timeout_config = conf.timeout_config.as_ref().cloned().unwrap_or_default();
                     let sleep_impl = conf.sleep_impl.clone();
-                    let mut client = #{aws_hyper}::Client::new(conn)
-                        .with_retry_config(retry_config.into())
-                        .with_timeout_config(timeout_config);
-
-                    client.set_sleep_impl(sleep_impl);
+                    let mut builder = #{aws_smithy_client}::Builder::new()
+                        .connector(conn)
+                        .middleware(#{aws_hyper}::AwsMiddleware::new());
+                    builder.set_retry_config(retry_config.into());
+                    builder.set_timeout_config(timeout_config);
+                    if let Some(sleep_impl) = sleep_impl {
+                        builder.set_sleep_impl(Some(sleep_impl));
+                    }
```

Comment:
Isn't this equivalent to `builder.set_sleep_impl(sleep_impl);`? Or is the builder in this case maintaining a tri-state? If the latter, this probably deserves a comment.